### ISSUE TITLE
DatabaseView throws error when starting with WITH statement

### ISSUE THREAD

**Issue by themadmrj** [vitusortner/floor#440]
`@DatabaseView("WITH lastMessages as (SELECT *, MAX(creationTimestamp) creationTimestamp FROM chat_message GROUP BY chatid) SELECT chat.*, lastMessages.* FROM chat INNER JOIN lastMessages ON chat.id = lastMessages.chatid")`
`class ChatWithLastMessage { ... }`

This query throws an error: `There is no SELECT query defined on the database view ChatWithLastMessage.`

Rewriting to `@DatabaseView("SELECT chat.*, (SELECT *, MAX(creationTimestamp) creationTimestamp FROM chat_message GROUP BY chatid) as lastMessages FROM chat INNER JOIN lastMessages ON chat.id = lastMessages.chatid")` seems to work.

**Comment by vitusortner** [comment]
Thanks for your report. Common table expressions (temporary views created with "WITH") are currently not supported. I might be able to look into this in the next days. If you're interested in contributing, you can take a look at the `ViewProcessor` which currently checks if a view query starts with `SELECT`. You can find the relevant code here: https://github.com/vitusortner/floor/blob/1a49c0f3c3cc49e32846e15472ce13ecea2d6ce2/floor_generator/lib/processor/view_processor.dart#L49

A solution could be to allow queries staring with `WITH` that additionally contain at least one `SELECT`.

**Comment by themadmrj** [comment]
So I can basically just make a pull request changing:
```
if (query == null || !query.trimLeft().toLowerCase().startsWith('select')) {
  throw _processorError.missingQuery;
}
```
to
```
if (query == null || (!query.trimLeft().toLowerCase().startsWith('select') && !(query.trimLeft().toLowerCase().startsWith('with') && query.toLowerCase().contains('select')))) {
  throw _processorError.missingQuery;
}

?
```

**Comment by vitusortner** [comment]
Sounds perfect. Please don't forget to add tests 👌

**Comment by themadmrj** [comment]
> Sounds perfect. Please don't forget to add tests 👌

I'll give it a shot... I've never actually written tests :p

**Comment by vitusortner** [comment]
Closed by #443
